File: Application/src/main/java/com/shareyourproxy/app/fragment/SecondIntroductionFragment.java
package com.shareyourproxy.app.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.shareyourproxy.R;

import butterknife.Bind;
import butterknife.BindDimen;
import butterknife.BindString;
import butterknife.ButterKnife;

/**
 * Second introduction slide content.
 */
public class SecondIntroductionFragment extends BaseIntroductionFragment {
    @BindDimen(R.dimen.common_svg_xlarge)
    int logoSize;
    @BindString(R.string.slide_two_title)
    String introTitle;
    @BindString(R.string.slide_two_body)
    String introBody;
    @Bind(R.id.fragment_introduction_second_imageview)
    ImageView imageView;
    @Bind(R.id.fragment_introduction_second_textview)
    TextView textView;

    /**
     * Default Constructor.
     */
    public SecondIntroductionFragment() {
    }

    public static SecondIntroductionFragment newInstance() {
        return new SecondIntroductionFragment();
    }

    @Override
    public View onCreateView(
        LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_introduction_second, container, false);
        ButterKnife.bind(this, rootView);
        drawSlide(getActivity(), imageView, textView, R.raw.ic_guide_activity_slide2, logoSize,
            introTitle, introBody);
        return rootView;
    }

}

File: Application/src/main/kotlin/com/shareyourproxy/app/fragment/SecondIntroductionFragment.kt
package com.shareyourproxy.app.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.shareyourproxy.R
import com.shareyourproxy.R.id.fragment_introduction_second_imageview
import com.shareyourproxy.R.id.fragment_introduction_second_textview
import com.shareyourproxy.R.raw.ic_guide_activity_slide2
import com.shareyourproxy.util.ButterKnife.bindDimen
import com.shareyourproxy.util.ButterKnife.bindString
import com.shareyourproxy.util.ButterKnife.bindView

/**
 * Second introduction slide content.
 */
class SecondIntroductionFragment : BaseIntroductionFragment() {
    private val logoSize: Int by bindDimen(R.dimen.common_svg_xlarge)
    private val introTitle: String by bindString(R.string.slide_two_title)
    private val introBody: String by bindString(R.string.slide_two_body)
    private val imageView: ImageView by bindView(fragment_introduction_second_imageview)
    private val textView: TextView by bindView(fragment_introduction_second_textview)

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_introduction_second, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        drawSlide(activity, imageView, textView, ic_guide_activity_slide2, logoSize, introTitle, introBody)
    }
}
